### PR TITLE
Remove 4 instances of 'gutenberg' text domain from WordPress core

### LINKED ISSUE #37228: Gutenberg textdomain in WordPress 5.9 beta 2.
### Description

Multiple instanses of the incorrect textdomain 'gutenberg' in core now. Why so ignorant attitude, not the first time this happens? 

https://github.com/kebbet/wordpress-develop/blob/0cab817ba4ec57224c3e7048dbe57478851ca813/src/wp-includes/blocks/navigation-submenu.php#L229
https://github.com/kebbet/wordpress-develop/blob/0cab817ba4ec57224c3e7048dbe57478851ca813/src/wp-includes/blocks/navigation-submenu.php#L233

https://github.com/kebbet/wordpress-develop/blob/0cab817ba4ec57224c3e7048dbe57478851ca813/src/wp-includes/blocks/page-list.php#L181
https://github.com/kebbet/wordpress-develop/blob/0cab817ba4ec57224c3e7048dbe57478851ca813/src/wp-includes/blocks/page-list.php#L189

### Step-by-step reproduction instructions

Do a simple search for 'gutenberg' in wordpress-develop.

### Screenshots, screen recording, code snippet

_No response_

### Environment info

WordPress trunk.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by kebbet**
Closes #37228 

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/block-library/src/navigation-submenu/index.php [hunk 1]
```diff
@@ -188,6 +188,12 @@ function render_block_core_navigation_submenu( $attributes, $content, $block ) {
 		);
 	}
 
+	$arialabel = sprintf(
```
Comment (Mamaduka) [inline packages/block-library/src/navigation-submenu/index.php]: Nitpick: Maybe we should use underscore as a separator in name of this variable?

**Review by Mamaduka — APPROVED** [review]
Thanks for fixing this, @kebbet.

Changes look good, I just left a small comment, but it's not a blocker.

File: packages/block-library/src/navigation-submenu/index.php (hunk 1 above)
Comment (ocean90) [reply packages/block-library/src/navigation-submenu/index.php]: Yes, the variable should be `$aria_label`.

File: packages/block-library/src/navigation-submenu/index.php [hunk 2]
```diff
@@ -188,6 +188,12 @@ function render_block_core_navigation_submenu( $attributes, $content, $block ) {
...
+		/* translators: button label name */
+		__( '%s submenu' ),
```
Comment (ocean90) [inline packages/block-library/src/navigation-submenu/index.php]: The translator comment is not really helpful as it doesn't explain what the `%s` placeholder is for. It's also not so easy to translate. My suggestion:

```suggestion
		/* translators: Accessibility text. %s: Parent page title. */
		__( 'Submenu of %s' ),
```

**Review by ocean90 — CHANGES_REQUESTED** [review]

File: packages/block-library/src/navigation-submenu/index.php (hunk 1 above)
Comment (kebbet) [reply packages/block-library/src/navigation-submenu/index.php]: Updated in the 2 latest commits! Thanks for pointing out!

File: packages/block-library/src/navigation-submenu/index.php (hunk 2 above)
Comment (kebbet) [reply packages/block-library/src/navigation-submenu/index.php]: Was struggling with it myself, thanks for suggestion and the review! Updated both cases to the same now.

**Comment by Mamaduka** [comment]
Hi, @ocean90

Can we merge this PR?

**Review by aristath — APPROVED** [review]

**Comment by aristath** [comment]
All comments were addressed. I reviewed the commit and it looks good, fixing the issue at hand. No failing tests, so I went ahead and merged it 👍

**Pull request merged**